packages/ic-http-certification/src/utils/response_header.rs

```diff
@@ -0,0 +1,95 @@
+/// * `witness` - A pruned merkle tree revealing the relevant certification for the current response.
+///     Created using [`HttpCertificationTree::witness()`](crate::HttpCertificationTree::witness).
+/// * `expr_path` - An expression path for the current response informing the HTTP Gateway where the
+///     relevant certification is present in the merkle tree. Created using
+///     [`HttpCertificationPath::to_expr_path()`](crate::HttpCertificationPath::to_expr_path).
+///
+/// # Example
+///
+/// ```
+/// use ic_http_certification::{HttpCertification, HttpRequest, HttpResponse, DefaultCelBuilder, DefaultResponseCertification, HttpCertificationTree, HttpCertificationTreeEntry, HttpCertificationPath, CERTIFICATE_EXPRESSION_HEADER_NAME, CERTIFICATE_HEADER_NAME, utils::add_certificate_header};
+///
+/// let cel_expr = DefaultCelBuilder::full_certification().build();
+///
+/// let request = HttpRequest::get("/index.html?foo=a&bar=b&baz=c").build();
+///
+/// let mut response = HttpResponse::builder()
+///     .with_headers(vec![(CERTIFICATE_EXPRESSION_HEADER_NAME.to_string(), cel_expr.to_string())])
+///     .build();
+///
+/// let request_url = "/example.json";
+/// let path = HttpCertificationPath::exact(request_url);
+/// let expr_path = path.to_expr_path();
+///
+/// let certification = HttpCertification::full(&cel_expr, &request, &response, None).unwrap();
+/// let entry = HttpCertificationTreeEntry::new(&path, &certification);
+///
+/// let mut http_certification_tree = HttpCertificationTree::default();
+/// http_certification_tree.insert(&entry);
+///
+/// // this should normally be retrieved using `ic_cdk::api::data_certificate()`.
+/// let data_certificate = vec![1, 2, 3];
+///
+/// let witness = http_certification_tree.witness(&entry, request_url).unwrap();
+/// add_certificate_header(
+///     data_certificate,
+///     &mut response,
+///     &witness,
+///     &expr_path
+/// );
+///
+/// assert_eq!(
+///     response.headers(),
+///     vec![
+///         (CERTIFICATE_EXPRESSION_HEADER_NAME.to_string(), cel_expr.to_string()),
+///         (
+///             CERTIFICATE_HEADER_NAME.to_string(),
+///             "certificate=:AQID:, tree=:2dn3gwJJaHR0cF9leHBygwJMZXhhbXBsZS5qc29ugwJDPCQ+gwJYIFJ2k+R/YYbgGPADidRdRwDurH06HXACVHlTIVrv1q4WgwJYIPZhxTCrVVSCuQKpNIckLOog7Q9SpfZ/0AODejmxpJ7egwJYIM7zUx3VibIaHEUF14Kx813l3Xlilg43Y5uGaABAA/i9ggNA:, expr_path=:2dn3g2lodHRwX2V4cHJsZXhhbXBsZS5qc29uYzwkPg==:, version=2".to_string(),
+///         ),
+///     ]
+/// );
+/// ```
+pub fn add_certificate_header(
+    data_certificate: Vec<u8>,
+    response: &mut HttpResponse,
+    witness: &HashTree,
+    expr_path: &[String],
+) {
+    let witness = cbor_encode(witness);
+    let expr_path = cbor_encode(&expr_path);
+
+    response.add_header((
+        CERTIFICATE_HEADER_NAME.to_string(),
+        format!(
+            "certificate=:{}:, tree=:{}:, expr_path=:{}:, version=2",
```

Comment:
since this function has the certification version hard coded, maybe it would be prude to mention this at least in the documentation, or even in the function name?  (e.g. `add_certificate_v2_header`?)
What would be the plan if we want to introduce v3?  A breaking change or a new API?

Reply:
That's a good point. It's not clear that this function only applies to v2.

I like the suggestion to rename the function. I went with `add_v2_certificate_header`, which felt a little nicer to say to me but I cannot explain why so I'm fine to change to `add_certificate_v2_header` if you feel strongly about it.

> What would be the plan if we want to introduce v3? A breaking change or a new API?

It's really difficult to say until we know what will actually change in v3. If it's simply the `version` part of the header then we could easily parameterize the function, but if somehow the structure of the header changes then maybe an entirely new function would be better.